### DISCUSSION TITLE
PerfGraphOutput

### DISCUSSION THREAD

**Discussion by selarem** [idaholab/moose#26632] [Q&A Modules: General] [answered]
hello,
I would like to print the perfolog at the end of  each timestep but with this :
[pgraph]
    type = PerfGraphOutput
    execute_on = 'initial final'   
    level = 2                     
    heaviest_branch = true       
    heaviest_sections = 7         # Default is 0
    interval =1
  []

I have only at the beginning of the first step.  Is execute_on = 'initial final'   refers to the start and end of the  entire computation ?  
regards,

**Comment by GiudGiud** [comment, marked as answer]
Hello

Yes exactly.

For timestep begin and end it s execute_on = timestep_begin/end respectively 

Guillaume
